image_processing/main.py

```diff
@@ -411,3 +410,102 @@ def upload_processed_images(bucket_name, source_file_folder):
+        source_file_name: path of file to upload
+    """
+
+    storage_client = storage.Client()
+
+    bucket = storage_client.bucket(bucket_name)
+    blob = bucket.blob(destination_blob_name)
+    blob.upload_from_filename(source_file_name)
+    return destination_blob_name
+
+
+def trigger_detect_and_mask(event, context):
+    """
+    Trigger function to detect and mask an image storaged
+    in the sie-raw-images bucket. Saves the masked image in
+    sie-masked-images.
+
+    Args:
+        event (dict):  The dictionary with data specific to this type of event.
+                        The `data` field contains a description of the event in
+                        the Cloud Storage `object` format described here:
+                        https://cloud.google.com/storage/docs/json_api/v1/objects#resource
+        context (google.cloud.functions.Context): Metadata of triggering event.
+    Returns:
+        str; error or success message
+    """
+
+    bucket_name = event["bucket"]  # sie-raw-images bucket
+    cloud_storage_prefix = "gs://" + bucket_name + "/"
+    cloud_download_from = event["name"]  # file path
+
+    # extract file name from path eg: folder/image.jpg
+    file_name = cloud_download_from.split("/")[-1]
+
+    # local directory to store image
+    cloud_download_to = "/" + constants.TEMP_DIR + "/" + file_name
+
+    # fetching image from this URI
+    uri = cloud_storage_prefix + cloud_download_from
+
+    try:
+        # check for face
+        results = face_detection(uri)
+
+        # download images from bucket sie-raw-images to create a mask
+        download_image(bucket_name, cloud_download_from, cloud_download_to)
+        print(f"Image {cloud_download_from} downloaded to {cloud_download_to}")
+
+        # face annotations
+        topLeft_x, topLeft_y, bottomRight_x, bottomRight_y = (
+            results[0],
+            results[1],
+            results[2],
+            results[3],
+        )
+        mid_eyes, nose, left_ear, right_ear, chin = (
+            results[4],
+            results[5],
+            results[6],
+            results[7],
+            results[8],
+        )
+
+        # process and create a mask
+        cloud_upload_from = process_img(
+            cloud_download_to,
+            topLeft_x,
+            topLeft_y,
+            bottomRight_x,
+            bottomRight_y,
+            mid_eyes,
+            nose,
+            left_ear,
+            right_ear,
+            chin,
+        )
+
+        # Upload masked image to sie-masked-images bucket
+        participant_id = os.path.splitext(event["name"])[0]  # remove file ext
+        cloud_upload_to = f"{participant_id}/{constants.PROCESSED_IMAGE}"
+        upload_image("sie-masked-images", cloud_upload_from, cloud_upload_to)
+        print("Processed image saved at: " + cloud_upload_to)
+
+    except exceptions.InvalidFaceImage as err:
+        traceback.print_exception(*sys.exc_info())
```

Comment:
i'm curious here ~ why did you use `traceback.print_exception(*sys.exc_info())`?

Reply:
Without this, there were no error messages printing in the logs. This line will print a comprehensive stack trace which helps in debugging.

Reply:
TIL! Thanks!